API/Extensions/FileUploader.cs

```diff
@@ -0,0 +1,134 @@
+        string RemoveSpecialCharacters(string str);
+
+
+        /// <summary>
+        /// Uploads single file
+        /// </summary>
+        /// <param name="file"></param>
+        /// <param name="fileName"></param>
+        /// <returns> path of file location </returns>
+        Task<string> UploadSingleFile(IFormFile file, string fileName);
+
+        /// <summary>
+        /// Method deletes the file from the file server
+        /// </summary>
+        /// <param name="file"></param>
+        void DeleteFile(File file);
+
+    }
+
+    /// <summary>
+    /// Class which is responsible for uploading files
+    /// </summary>
+    public class FileUploader : IFileUploader
+    {
+
+        private static readonly string UploadPath = Directory.GetCurrentDirectory() + "\\files\\";
+
+        /// <summary>
+        /// Removes special characters for string to avoid problems
+        /// </summary>
+        /// <param name="str"></param>
+        /// <returns> String without special characters</returns>
+        public string RemoveSpecialCharacters(string str)
+        {
+            StringBuilder sb = new StringBuilder();
+            foreach(char c in str)
+            {
+                if((c >= '0' && c <= '9') || (c >= 'A' && c <= 'Z') || (c >= 'a' && c <= 'z') || c == '.' || c == '_')
+                {
+                    sb.Append(c);
+                }
+            }
+            return sb.ToString();
+        }
+
+
+        /// <summary>
+        /// Uploads single file
+        /// </summary>
+        /// <param name="file"> File to upload </param>
+        /// <param name="fileName"> Name of file </param> 
+        /// <returns> path of file location </returns>
+        public async Task<string> UploadSingleFile(IFormFile file, string fileName)
+        {
+            try
+            {
+                if(!System.IO.File.Exists(UploadPath + fileName))
+                {
+                    await using(Stream sourceStream = file.OpenReadStream())
+                    {
+                        await using(FileStream destinationStream = System.IO.File.Create(UploadPath + fileName))
+                        {
+                            await sourceStream.CopyToAsync(destinationStream);
+                        }
+                    }
+
+                    return UploadPath + fileName;
+                }
+
+                throw new FileExistException(fileName);
+            } catch(Exception e)
+            {
+                Log.Logger.Error(e, "Unexpected error");
+                throw e;
+            }
+        }
+
+        /// <summary>
+        /// Method deletes the file from the file server
+        /// </summary>
+        /// <param name="file"></param>
+        /// <returns> Bool which tells if file is deleted succesfully or not </returns>
+        public void DeleteFile(File file)
+        {
+            if(System.IO.File.Exists(Path.Combine(UploadPath, file.Name)))
```

Comment:
future iteration should check if the user making the request 'owns' the file to be deleted.